ocaml/lambda/lambda.ml

```diff
@@ -1400,16 +1400,21 @@ let build_substs update_env ?(freshen_bound_variables = false) s =
     | Levent (lam, evt) ->
         let old_env = evt.lev_env in
         let env_updates =
-          let find_in_old id = Env.find_value (Path.Pident id) old_env in
+          let find_in_old id =
+            let vd = Env.find_value (Path.Pident id) old_env in
+            let vd = {vd with val_modalities = Mode.Modality.Value.id} in
+            let mode = Mode.Value.max |> Mode.Value.disallow_right in
```

Comment:
The modalities returned by `Env.find_value` will have `val_modalities` invalidated, so we are not dropping useful information. Essentially I'm assuming here that modes are not important here, and thus weakening it to `max` for conservative soundness.

I tried to preserve the mode information but it's complex. We lookup values in the environment (which potentially cross locks) and adds them to a new environment (without locks). We shouldn't run the locks, because it's not "real usage". But  it's just wrong to use the unlocked mode directly at L1406. If you ask me, I think the existing code is already wrong.

Fortunately, the `new_env` here is only used when printing in the debugger. So the sigficance is low.

Reply:
Ah, gotcha. Okay, yeah, it would be good to comment here that `val_modalities` would otherwise be undefined and we just need something that the debugger will be able to display